### PR TITLE
Add search test

### PR DESCRIPTION
**Pull request by microbit-robert**
Adds a basic test to open the search modal, input a search term and navigate to the first result.

### CODE REVIEW

File: src/e2e/app.ts [hunk 1]
```diff
@@ -790,6 +790,24 @@ export class App {
...
+    const searchField = await document.findByRole("textbox", {
+      name: "Search",
+    });
+    await searchField.type(searchText);
+    const modalDialog = await document.findByRole("dialog");
+    // const result = await modalDialog.getAllByRole("button");
+    const result = await modalDialog.findAllByRole("heading", {
+      level: 3,
+    });
+    await result[0].click();
```
Comment (microbit-matt-hillsdon) [inline src/e2e/app.ts:808]: I think it's not clear from the method name that this clicks on the first result.

Maybe split into two?

**Review by microbit-matt-hillsdon — CHANGES_REQUESTED** [review]
Looks good. One suggestion above.

File: src/e2e/app.ts (hunk 1 above)
Comment (microbit-robert) [reply src/e2e/app.ts:811]: [`cf28c83`](https://github.com/microbit-foundation/python-editor-next/pull/474/commits/cf28c83a6e090bc5b244e6327e4eade9b7356502)

**Review by microbit-matt-hillsdon — APPROVED** [review]

File: src/e2e/app.ts [hunk 2]
```diff
@@ -107,6 +107,17 @@ export class App {
       await dialog.accept();
     });
 
+    const logsPath =
+      "reports/screenshots/" + expect.getState().currentTestName + ".txt";
+    page.on("console", (msg) => {
+      fs.appendFile(logsPath, msg.text(), (err) => {
+        if (err) {
+          // Log file error.
+          console.log("Log file error: ", err.message);
```
Comment (microbit-matt-hillsdon) [inline src/e2e/app.ts:116]: I think this is fine to be best effort with a logged error, but let's at least do console.error.

File: src/e2e/app.ts [hunk 3]
```diff
@@ -107,6 +107,17 @@ export class App {
       await dialog.accept();
     });
 
+    const logsPath =
+      "reports/screenshots/" + expect.getState().currentTestName + ".txt";
+    page.on("console", (msg) => {
+      fs.appendFile(logsPath, msg.text(), (err) => {
```
Comment (microbit-matt-hillsdon) [inline src/e2e/app.ts:113]: If you look at the published files they seem to have all the text on one line.
Perhaps we need to add a newline character here?

File: src/e2e/app.ts [hunk 4]
```diff
@@ -107,6 +107,17 @@ export class App {
       await dialog.accept();
     });
 
+    const logsPath =
+      "reports/screenshots/" + expect.getState().currentTestName + ".txt";
```
Comment (microbit-matt-hillsdon) [inline src/e2e/app.ts:111]: I think it'd now make more sense if "screenshots" was be e.g. "e2e" here and below (ideally using a common constant).

**Review by microbit-matt-hillsdon — CHANGES_REQUESTED** [review]
Some tweaks to the logging needed.

File: src/e2e/app.ts (hunk 4 above)
Comment (microbit-robert) [reply src/e2e/app.ts]: [`109f052`](https://github.com/microbit-foundation/python-editor-next/pull/474/commits/109f052b09bee1277305477bdc62a428562bf793)

File: src/e2e/app.ts (hunk 3 above)
Comment (microbit-robert) [reply src/e2e/app.ts]: [`109f052`](https://github.com/microbit-foundation/python-editor-next/pull/474/commits/109f052b09bee1277305477bdc62a428562bf793)

File: src/e2e/app.ts (hunk 2 above)
Comment (microbit-robert) [reply src/e2e/app.ts]: [`109f052`](https://github.com/microbit-foundation/python-editor-next/pull/474/commits/109f052b09bee1277305477bdc62a428562bf793)

File: src/e2e/app.ts (hunk 3 above)
Comment (microbit-matt-hillsdon) [reply src/e2e/app.ts]: Change looks good. Will just review the output then merge.

**Comment by microbit-matt-hillsdon** [comment]
Looks good now. Interesting to see we don't output screenshots for every case (edits e2e test) but I'll fix that separately.

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Pull request merged**